website_hr_job_unpublication/data/ir_cron.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2017 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record id="ir_cron_hr_job_auto_unpublish" model="ir.cron">
+        <field name="name">Hr Job Automatic Unpublication</field>
+        <field name="user_id" ref="base.user_root" />
+        <field name="interval_type">work_days</field>
```

Comment:
If the unpublish date is a datetime field, I guess the cron should be executed every hours.